.github/workflows/test-reproducibility.yml

```diff
@@ -0,0 +1,40 @@
+name: Reproducibility tests
+
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  run-repro-tests:
+    name: Run reproducibility tests for different algorithms
```

Comment:
Very nice work @ayushbaid. Was curious about the motivation for a separate test workflow, vs. incorporating into the main test workflow.

Reply:
Having it in a different workflow will enable the unit-tests (must be successful for the PR) to be tracked better, as the reproducibility tests might introduce failures. Also, we can configure reproducibility tests to not run every time if we want that in the future.

Reply:
Makes sense. Marking as resolved.